lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -681,10 +681,21 @@ class SPIRVBinary : public SPIRVInstTemplateBase {
              "Invalid type for bitwise instruction");
       assert((Op1Ty->getIntegerBitWidth() == Op2Ty->getIntegerBitWidth()) &&
              "Inconsistent BitWidth");
+    } else if (isBinaryPtrOpCode(OpCode)) {
+      assert((Op1Ty->isTypePointer() && Op2Ty->isTypePointer()) &&
+             "Invalid type for Binary Ptr instruction");
+      assert(static_cast<SPIRVTypePointer *>(Op1Ty)->getElementType() ==
+                 static_cast<SPIRVTypePointer *>(Op2Ty)->getElementType() &&
+             "Invalid types for Binary Ptr instruction");
     } else {
       assert(0 && "Invalid op code!");
```

Comment:
It's a nice occasion to improve diagnostics in this place and add precise info about op code that is not supported. I've once hit this exactly line when SPIRV Backend generated code for missed OpPtrEqual/NotEqual and spent some time on debugging. To avoid this in future we may want to improve diagnostics right in this place and give a user-friendly description of what exactly SPIRV Translator doesn't have support for.
@MrSidims @asudarsa what do you think? ^

Reply:
Agree with Slava

Reply:
I think the subsequent change from @vmaksimo does provide more information. One issue I see here is our reliance on 'assert'. Should we try to provide an error mechanism which will stop translation in such cases?

Thanks